hack/release/common.sh

```diff
@@ -66,136 +53,165 @@ authenticatePrivateRepo() {
+
+  oci_repo="${1}"
+  helm_chart="${2}"
+  version="${3}"
+  commit_sha="${4}"
+  build_date="${5}"
+
+  ah_config_file_name"${helm_chart}/artifacthub-repo.yaml"
+  helm_chart_artifact="${helm_chart}-${version}.tgz"
 
-    if [[ $(releaseType "$RELEASE_VERSION") == "$RELEASE_TYPE_SNAPSHOT" ]]; then
-      echo "${CURRENT_MAJOR_VERSION}-${RELEASE_VERSION}"
-    fi
+  yq e -i ".appVersion = \"${version}\"" "charts/${helm_chart}/Chart.yaml"
+  yq e -i ".version = \"${version}\"" "charts/${helm_chart}/Chart.yaml"
+
+  cd charts
+  [[ -s "${ah_config_file_name}" ]] && oras push "${oci_repo}:artifacthub.io" --config /dev/null:application/vnd.cncf.artifacthub.config.v1+yaml "${ah_config_file_name}:application/vnd.cncf.artifacthub.repository-metadata.layer.v1.yaml"
+  helm dependency update "${helm_chart}"
+  helm lint "${helm_chart}"
+  helm package "${helm_chart}" --version "${version}"
+  helm push "${helm_chart_artifact}" "oci://${oci_repo}"
+  rm "${helm_chart_artifact}"
+  cd ..
+
+  helm_chart_digest="$(crane digest "${oci_repo}:${version}")"
+  cosignOciArtifact "${commit_sha}" "${build_date}" "${oci_repo}:${version}@${helm_chart_digest}"
 }
 
-buildDate(){
-    # Set the SOURCE_DATE_EPOCH and KO_DATA_DATE_EPOCH values for reproducable builds with timestamps
-    # https://ko.build/advanced/faq/
-    DATE_FMT="+%Y-%m-%dT%H:%M:%SZ"
-    SOURCE_DATE_EPOCH=$(git log -1 --format='%ct')
-    echo "$(date -u -r "${SOURCE_DATE_EPOCH}" $DATE_FMT 2>/dev/null)"
+cosignOciArtifact() {
+  local commit_sha build_date artifact
+
+  commit_sha="${1}"
+  build_date="${2}"
+  artifact="${3}"
+
+  cosign sign --yes -a commitSha="${commit_sha}" -a buildDate="${build_date}" "${artifact}"
 }
 
-cosignImages() {
-    cosign sign --yes \
-        -a GIT_HASH="$(git rev-parse HEAD)" \
-        -a GIT_VERSION="${RELEASE_VERSION}" \
-        -a BUILD_DATE="$(buildDate)" \
-        "${CONTROLLER_IMG}"
+dateEpoch() {
+  git log -1 --format='%ct'
 }
 
-publishHelmChart() {
-    CHART_NAME=$1
-    RELEASE_VERSION=$2
-    RELEASE_REPO=$3
-    HELM_CHART_VERSION=$(helmChartVersion "$RELEASE_VERSION")
-    HELM_CHART_FILE_NAME="${CHART_NAME}-${HELM_CHART_VERSION}.tgz"
-    AH_CONFIG_FILE_NAME="${CHART_NAME}/artifacthub-repo.yaml"
-
-    cd charts
-    [[ -s "${AH_CONFIG_FILE_NAME}" ]] && oras push "${RELEASE_REPO}:artifacthub.io" --config /dev/null:application/vnd.cncf.artifacthub.config.v1+yaml "${AH_CONFIG_FILE_NAME}:application/vnd.cncf.artifacthub.repository-metadata.layer.v1.yaml"
-    helm dependency update "${CHART_NAME}"
-    helm lint "${CHART_NAME}"
-    helm package "${CHART_NAME}" --version "${HELM_CHART_VERSION}"
-    helm push "${HELM_CHART_FILE_NAME}" "oci://${RELEASE_REPO}"
-    rm "${HELM_CHART_FILE_NAME}"
-    cd ..
-
-    cosignHelmChart "${RELEASE_REPO}/${CHART_NAME}" "${HELM_CHART_VERSION}"
+buildDate() {
+  local date_epoch
+
+  date_epoch="${1}"
+
+  date -u -r "${date_epoch}" "+%Y-%m-%dT%H:%M:%SZ" 2>/dev/null
```

Comment:
What's the reason/need to change the command that is being called to produce the buildDate from before?

Reply:
The command hasn't changed but I've refactored it to remove the unnecessary `echo` and extract the epoch as the same command was being used in multiple places.